### PR TITLE
feat(auth): add authentication using github as OAuth2 provider

### PR DESCRIPTION
**Pull request by SteveVaknin**
Title is pretty clear

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -1,17 +1,71 @@
...
+const session = require("express-session");
 const pino = require('express-pino-logger')();
+const proxy = require("express-http-proxy");
+const passport = require('passport');
+// const OAuth2Strategy = require('passport-oauth2'); will be needed after whiten
+const loginRouter = require('./routes/auth/auth');
+
+const GitHubStrategy = require('passport-github').Strategy;
+
+
```
Comment (adirsiton) [inline server/index.js]: Remove those lines

File: server/index.js [hunk 2]
```diff
@@ -1,17 +1,71 @@
...
+);
+app.use(passport.initialize());
+app.use(passport.session());
+
+passport.use('OAuth2', new GitHubStrategy({
+  // authorizationURL: 'https://github.com/login/oauth/authorize', will be needed after whiten
+  // tokenURL: 'https://github.com/login/oauth/access_token', will be needed after whiten
+  clientID: "2c29d26a0e4bd9ee8c35",
+  clientSecret: "28add6baf72c63bcca938dc94660ab92c3374af4",
+  callbackURL: "http://localhost:3001/auth/callback",
```
Comment (adirsiton) [inline server/index.js]: Shouldn't it be configures in `.env`?

File: server/index.js [hunk 3]
```diff
@@ -1,17 +1,71 @@
...
+  }
+});
 
 app.get('/api/greeting', (req, res) => {
   const name = req.query.name || 'World';
   res.setHeader('Content-Type', 'application/json');
   res.send(JSON.stringify({ greeting: `Hello ${name}!` }));
 });
 
+app.get("*", proxy("http://localhost:3000"));
```
Comment (adirsiton) [inline server/index.js]: Same as last comment

**Comment by SteveVaknin** [comment]
Dont forget to update you `.env` file, i will send the secrets in teams

**Pull request merged**